components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -86,7 +86,14 @@ public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bod
                 getPublicClientSupportedGrantTypesList();
         List grantTypes = bodyParams.get(GRANT_TYPE);
 
-        if (!publicClientSupportedGrantTypes.isEmpty() && grantTypes != null) {
+        if (publicClientSupportedGrantTypes.isEmpty()) {
+            if (log.isDebugEnabled()) {
```

Comment:
We don't need `isDebugEnabled` check since there are no string concatenations

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2161/commits/cbe612b67ce71393be554659b387d991c78dd89e